src/soc/hw/noc/test/test_noc_buffer.py

```diff
@@ -0,0 +1,343 @@
+"""
+Test the module noc_buffer.
+"""
+
+import random
+
+import cocotb
+from cocotb.clock import Clock
+from cocotb.triggers import RisingEdge, FallingEdge
+from cocotb.result import TestFailure
+
+# Mirroring expected contents of the FIFO
+_buffer_flits = []
+_buffer_last = []
+
+@cocotb.test()
+def test_noc_buffer(dut):
+    """
+    Test the module noc_buffer.
+    """
+
+    # Read the parameters back from the DUT to set up our model
+    width = dut.FLIT_WIDTH.value.integer
+    depth = dut.DEPTH.value.integer
+    fullpacket = dut.FULLPACKET.value.integer
+    dut._log.info("NoC buffer for a flit width of {} bits and a depth of {}. ".format(width, depth) +
+                  "FULLPACKET: {}".format(fullpacket))
+
+    # Setup clock
+    cocotb.fork(Clock(dut.clk, 10, "ns").start())
+
+    # Reset
+    dut._log.info("Resetting DUT")
+    dut.rst <= 1
+
+    dut.in_flit <= 0
+    dut.in_valid <= 0
+    dut.in_last <= 0
+    dut.out_ready <= 0
+
+    for _ in range(2):
+        yield RisingEdge(dut.clk)
+    dut.rst <= 0
+
+    # Special test case to test FULLPACKET behavior
+    if fullpacket == 1 and width == 8 and depth == 4:
+        dut._log.info("Run 'FULLPACKET' test case..")
+        yield fullpacket_test_case(dut)
+    # Otherwise, run normal read/write test if fullpacket == 0
+    elif fullpacket == 0:
+        # Start read and write processes
+        write_thread = cocotb.fork(write_buffer(dut, dut.clk, width))
+        read_thread = cocotb.fork(read_buffer(dut, dut.clk))
+
+        # Wait for read/write to finish. Read only finishes if all required data
+        # has been obtained, i.e. it implicitly waits for write as well.
+        yield read_thread.join()
+    else:
+        dut._log.info("No test defined for current parameter combination.")
```

Comment:
nit: what is the "current" parameter configuration? Either add the parameters to the message, or leave the message out, as it isn't overly useful as it is.